src/query/storage/m3/cluster_resolver.go

```diff
@@ -160,14 +166,14 @@ func resolveClusterNamespacesForQueryLogicalPlan(
 	unaggregated := resolveUnaggregatedNamespaceForQuery(now, start, ns, opts)
 	if unaggregated.satisfies == fullySatisfiesRange {
 		return consolidators.NamespaceCoversAllQueryRange,
-			ClusterNamespaces{unaggregated.clusterNamespace},
+			[]resolvedNamespace{resolved(unaggregated.clusterNamespace)},
```

Comment:
nit: maybe have a similar type `resolvedNamespaces` like we did with `ClusterNamespaces` ?

Reply:
OK. Even though I don't like that it hides the fact that it is a slice in places like
```go
make(resolvedNamespaces, 0, size)
```